doc/source/user_guide/mapdl.rst

```diff
@@ -66,60 +132,158 @@ method. Here is an example:
+
+generates the following input for MAPDL:
 
 .. code:: apdl
 
-    ! SELECT NODES AT Z = 10 TO APPLY DISPLACEMENT
-    *CREATE,DISP
+    NSEL,ALL   
     NSEL,R,LOC,Z,10
-    D,ALL,UZ,ARG1
-    NSEL,ALL
-    /OUT,SCRATCH
-    SOLVE
-    *END
 
-    ! Call the function
-    *USE,DISP,-.032
-    *USE,DISP,-.05
-    *USE,DISP,-.1
+which is executed with a :func:`Mapdl.input() <ansys.mapdl.core.Mapdl.input>` call.
 
-It should be written as follows:
+Because of the non-interactive context not running all the commands until the end,
+you might find issues interacting inside it, with Python for instance.
+For example, running Python commands such as
+:func:`Mapdl.get_array() <ansys.mapdl.core.Mapdl.get_array>`
+inside the context can give you out-of-sync responses.
+The following code snippet is a demonstration of this kind of problem:
 
 .. code:: python
 
-    def DISP(
-        ARG1="",
-        ARG2="",
-        ARG3="",
-        ARG4="",
-        ARG5="",
-        ARG6="",
-        ARG7="",
-        ARG8="",
-        ARG9="",
-        ARG10="",
-        ARG11="",
-        ARG12="",
-        ARG13="",
-        ARG14="",
-        ARG15="",
-        ARG16="",
-        ARG17="",
-        ARG18="",
-    ):
-        mapdl.nsel("R", "LOC", "Z", 10)  # SELECT NODES AT Z = 10 TO APPLY DISPLACEMENT
-        mapdl.d("ALL", "UZ", ARG1)
-        mapdl.nsel("ALL")
-        mapdl.run("/OUT,SCRATCH")
-        mapdl.solve()
+    # Create some keypoints
+    mapdl.clear()
+    mapdl.k(1, 0, 0, 0)
+    mapdl.k(2, 1, 0, 0)
+
+    with mapdl.non_interactive:
+        mapdl.k(3, 2, 0, 0)
+        klist_inside = mapdl.get_array("KP", item1="KLIST")
+        # Here is where PyMAPDL sends the commands to the MAPDL instance and execute 'mapdl.k(3,2,0,0)' (`K,3,2,0,0`
+
+    klist_outside = mapdl.get_array("KP", item1="KLIST")
+
+    assert klist_inside != klist_outside  # Evaluates to true
+
+In the preceding script, the values obtained by
+:func:`Mapdl.get_array() <ansys.mapdl.core.Mapdl.get_array>` are different:
+
+.. code:: pycon
+
+    >>> print(klist_inside)
+    array([1., 2.])
+    >>> print(klist_outside)
+    array([1., 2., 3.])
 
+This is because the first :func:`Mapdl.get_array() <ansys.mapdl.core.Mapdl.get_array>`
+call is executed actually *before* the :func:`Mapdl.k() <ansys.mapdl.core.Mapdl.k>` call.
 
-    DISP(-0.032)
-    DISP(-0.05)
-    DISP(-0.1)
+It is recommend to not retrieve any data in a pythonic way from the MAPDL instance while in the
+:meth:`non_interactive context <ansys.mapdl.core.Mapdl.non_interactive>`.
+Being aware of this kind of behavior and how
+:meth:`non_interactive context <ansys.mapdl.core.Mapdl.non_interactive>`
+works is crucial for an advanced usage of PyMAPDL.
+
+
+MAPDL macros
+------------
+Note that macros created within PyMAPDL (rather than loaded from
+a file) do not appear to run correctly. For example, here is the macro
+``DISP`` created using the ``*CREATE`` command within APDL and within PyMAPDL:
```

Comment:
```suggestion
a file) do not appear to run correctly. For example, here is the ``DISP``
macro created using the ``*CREATE`` command within APDL and within PyMAPDL:
```
The noun describing the object type should follow what is tagged as a code entity.